README.md

```diff
@@ -50,7 +50,87 @@ Click on the R:(note attached) button to launch R in the terminal. You can then
     <img src="screenshots/running_r_code.png" width="80%">
 </p>
 
-
+## R Contribution Workflow
+
+### Build Setup
+1. Environment Variables
+    - We have environment variables for setting path to install and build R and source code. And also for R source code version.
+    - The path ENV variable for R Build and R Source code are BUILDDIR and TOP_SRCDIR respectively. And for R source code version it is R_VERSION.
+    - The environment variables are set in the codespace image and are available when the codespace starts.
+    - The R_VERSION_BUILD_DIR has BUILDDIR path as base path and R_VERSION is added as suffix.
+      
+      ![image](https://github.com/r-devel/r-dev-env/assets/72031540/6b6cb661-cde3-4c69-845a-a6a281c18ec2)
+2. svn checkout
+   - The svn checkout cmd lets us create working of a repository with specific tag/branch.
+   - Example:
+    ```bash
+     svn checkout https://svn.r-project.org/R/tags/$R_VERSION/ "$R_VERSION_SOURCE_DIR"
+     ```
+   - Here we are checking out R source with specific version and the checkout files and folders will be stored inside R_VERSION_SOURCE_DIR directory.
+   - Output : We get file structure something like this after checking out R source code from R svn repository.
+     ![image](https://github.com/r-devel/r-dev-env/assets/72031540/5c560a5e-a40d-4c68-9013-d6eec7327020)
+
+3. cd to BUILDDIR
+   - We need to change our directory to R build directory(BUILDDIR) to build and configure our R source code.
+   - ```bash
+     cd $R_VERSION_BUILD_DIR
+     ```
+   - The R_VERSION_BUILD_DIR has BUILDDIR path as base path and R_VERSION is added as suffix.
+5. configure source code
+   - After we change directory to BUILDDIR we can build R and source code.
+   - ```bash
+     "$R_VERSION_SOURCE_DIR/configure" --enable-R-shlib --without-recommended-packages
+     make
+     sudo make install
+     ```
+   - The configure cmd help us build R and files and folders are stored inside BUILDDIR directory.
+   - Output : We get file structure something like this after using configure command.
+     
+     ![image](https://github.com/r-devel/r-dev-env/assets/72031540/b6279710-1176-4c56-9a09-7c41c582e5f8)
+
+6. After having configured setup, we can now make changes in source code and make our contributions.
+
+### Contribution Workflow
+1. R-org uses bugzilla for bug tracking and svn as a version control system.
+2. We can find listed and known bugs on bugzilla. Link : https://bugs.r-project.org/buglist.cgi?bug_status=__open__
+3. Example Contribution Workflow using DevContainer:
+   -  We will use bug [17616](https://bugs.r-project.org/show_bug.cgi?id=17616)
+   -  To confirm the bug we will click on `R:(not attach)` option which is in the bottom right of our R-dev codespace. It will open R terminal for us.
```

Comment:
```suggestion
   -  To start working in R we will click on `R:(not attach)` option which is in the bottom right of our R-dev codespace. It will open R terminal for us.
```